modules/pstudioBidAdapter.js

```diff
@@ -0,0 +1,439 @@
+        publisher: {
+          name: site.publisher.name,
+          cat: site.publisher.cat,
+          domain: site.publisher.domain,
+        },
+      }),
+      ...(site.content && {
+        content: {
+          id: site.content.id,
+          episode: site.content.episode,
+          title: site.content.title,
+          series: site.content.series,
+          artist: site.content.artist,
+          genre: site.content.genre,
+          album: site.content.album,
+          isrc: site.content.isrc,
+          season: site.content.season,
+        },
+      }),
+      mobile: site.mobile,
+    };
+  }
+
+  if (app) {
+    appData = {
+      id: app.id,
+      name: app.name,
+      bundle: app.bundle,
+      domain: app.domain,
+      storeurl: app.storeurl,
+      cat: app.cat,
+      sectioncat: app.sectioncat,
+      pagecat: app.pagecat,
+      ver: app.ver,
+      privacypolicy: app.privacypolicy,
+      paid: app.paid,
+      ...(app.publisher && {
+        publisher: {
+          name: app.publisher.name,
+          cat: app.publisher.cat,
+          domain: app.publisher.domain,
+        },
+      }),
+      keywords: app.keywords,
+      ...(app.content && {
+        content: {
+          id: app.content.id,
+          episode: app.content.episode,
+          title: app.content.title,
+          series: app.content.series,
+          artist: app.content.artist,
+          genre: app.content.genre,
+          album: app.content.album,
+          isrc: app.content.isrc,
+          season: app.content.season,
+        },
+      }),
+    };
+  }
+
+  if (regs) {
+    regsData = { coppa: regs.coppa };
+  }
+
+  return cleanObject({
+    user: userData,
+    device: deviceData,
+    site: siteData,
+    app: appData,
+    regs: regsData,
+  });
+}
+
+function cleanObject(data) {
+  for (let key in data) {
+    if (typeof data[key] == 'object') {
+      cleanObject(data[key]);
+
+      if (isEmpty(data[key])) delete data[key];
+    }
+
+    if (data[key] === undefined) delete data[key];
+  }
+
+  return data;
+}
+
+function isVideoRequestValid(bidRequest) {
+  if (bidRequest.mediaTypes.video) {
+    const { w, h, playerSize, mimes, protocols } = deepAccess(
+      bidRequest,
+      'mediaTypes.video',
+      {}
+    );
+
+    const areSizesValid =
+      (isNumber(w) && isNumber(h)) || validateSizes(playerSize);
+    const areMimesValid = isArray(mimes) && mimes.length > 0;
+    const areProtocolsValid =
+      isArray(protocols) && protocols.length > 0 && protocols.every(isNumber);
+
+    return areSizesValid && areMimesValid && areProtocolsValid;
+  }
+
+  return true;
+}
+
+function validateSizes(sizes) {
+  return (
+    isArray(sizes) &&
+    sizes.length > 0 &&
+    sizes.every(
+      (size) => isArray(size) && size.length === 2 && size.every(isNumber)
+    )
+  );
+}
+
+function getBidFloor(bid) {
+  if (bid.params.floorPrice) {
+    return bid.params.floorPrice;
+  }
+
+  if (!isFn(bid.getFloor)) {
+    return null;
+  }
+
```

Comment:
```suggestion
You should ensure there is no getFloor function before setting your floor in params:

function getBidFloor(bid) {
  if (!isFn(bid.getFloor)) {
    return bid.params.floorPrice ? bid.params.floorPrice : null;
  }

```

Reply:
ok you are right more important is function than parameter